src/main/java/sleppynavigators/studyupbackend/domain/chat/generator/action/ChatActionListGeneratorFactory.java

```diff
@@ -0,0 +1,43 @@
+package sleppynavigators.studyupbackend.domain.chat.generator.action;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+import sleppynavigators.studyupbackend.domain.chat.action.ChatAction;
+import sleppynavigators.studyupbackend.domain.event.Event;
+import sleppynavigators.studyupbackend.domain.event.EventType;
+
+@Component
+public class ChatActionListGeneratorFactory {
+
+    private final Map<EventType, ChatActionListGenerator<?>> generatorMap;
+    private final ChatActionListGenerator<?> fallbackGenerator = new FallbackChatActionListGenerator();
+
+    public ChatActionListGeneratorFactory(List<ChatActionListGenerator<? extends Event>> generators) {
+        this.generatorMap = generators.stream()
+                .collect(Collectors.toMap(
+                        ChatActionListGenerator::supportedEventType,
+                        generator -> generator));
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T extends Event> ChatActionListGenerator<T> get(T event) {
+        ChatActionListGenerator<?> generator = generatorMap.get(event.getType());
+        return (ChatActionListGenerator<T>) Objects.requireNonNullElse(generator, fallbackGenerator);
+    }
```

Comment:
```suggestion
    @Nullable
    @SuppressWarnings("unchecked")
    public <T extends Event> ChatActionListGenerator<T> get(T event) {
        return generatorMap.get(event.getType());
    }
```
fallback을 만들기보다는 명시적으로 null을 반환하면 어떨까요?

Reply:
fallback 대신 null을 반환할 경우 `ChatActionListGeneratorFactory`를 호출하는 쪽에서 null control을 해야 합니다. 동시에 해당하는 event type에 별도 generator가 없을 경우 어떤 동작을 해야 할지도 결정해야 하고요.

현재 `ChatActionListGeneratorFactory`를 Application Service에서 호출하고 있는데, generator가 없을 경우의 fallback behavior는 책임 경계 밖에 있다고 판단했습니다.